qiskit_experiments/library/tomography/basis/base_basis.py

```diff
@@ -49,108 +43,130 @@ def name(self) -> str:
+            qubits: the basis subsystems to return the index shape for.
 
         Returns:
-            The POVM matrix for the bases and outcome.
+            The shape of allowed values for the index on the specified qubits.
         """
 
-
-class BaseTomographyMeasurementBasis(BaseFitterMeasurementBasis):
-    """Abstract base class for tomography experiment measurement bases.
-
-    This class extends BaseFitterMeasurementBasis to include a
-    circuit definition of each basis index. It can be used to construct
-    tomography circuits for execution, as well as for tomography analysis
-    of measurement data.
-    """
-
     @abstractmethod
-    def circuit(self, index: Iterable[int]) -> QuantumCircuit:
-        """Return a composite rotation circuit to measure in basis
+    def circuit(self, index: Sequence[int], qubits: Sequence[int]) -> QuantumCircuit:
+        """Return the basis preparation circuit.
 
         Args:
             index: a list of basis elements to tensor together.
+            qubits: The physical qubit subsystems for the index.
 
         Returns:
-            the rotation circuit for the specified basis
+            The basis circuit for the specified index and qubits.
         """
 
 
-class BaseFitterPreparationBasis(ABC):
-    """Abstract base class for fitter preparation bases.
+class PreparationBasis(BaseBasis):
+    """Abstract base class for a tomography preparation basis.
 
-    This class defines the density matrix for each preparation basis
-    index. It is used by tomography fitters during tomography analysis.
-    """
+    Subclasses should implement the following abstract methods to
+    define a preparation basis:
 
-    def __init__(self, name: Optional[str] = None):
-        """Initialize a fitter preparation basis.
+    * The :meth:`circuit` method which returns the preparation
+      :class:`.QuantumCircuit` for basis element index on the specified
+      qubits.
 
-        Args:
-            name: Optional, name for the basis. If None the class
-                  name will be used.
-        """
-        self._name = name if name else type(self).__name__
+    * The :meth:`matrix` method which returns the density matrix prepared
+      by the bases element index on the specified qubits.
 
-    def __hash__(self):
-        return hash((type(self), self._name))
+    * The :meth:`index_shape` method which returns the shape of allowed
+      basis indices for the specified qubits, and their values.
 
-    def __eq__(self, value):
-        tup1 = (type(self), self.name)
-        tup2 = (type(value), getattr(value, "name", None))
-        return tup1 == tup2
+    * The :meth:`matrix_shape` method which returns the shape of subsystem
+      dimensions of the density matrix state on the specified qubits.
 
-    @property
-    def name(self) -> str:
-        """Return the basis name"""
-        return self._name
+    .. note::
+
+      The number of qubits and number of index elements do not necessarily
```

Comment:
I should remove this comment since its a bit confusing and isn't really necessary at the moment. But basically someone could build some specialized bases like a 2-qubit basis with a single index or something like that if they wanted to.